src/Content/ContentStorage.php

```diff
@@ -257,24 +241,24 @@ public function touchLanguage(string|null $lang): void
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
 	public function update(
-		string $version,
+		VersionId $versionId,
 		string|null $lang = null,
 		array $fields = []
 	): void {
 		$lang = $this->language($lang);
-		$this->ensureExistingVersion($version, $lang);
-		$this->handler->update($version, $lang, $fields);
+		$this->ensureExistingVersion($versionId, $lang);
+		$this->handler->update($versionId, $lang, $fields);
 	}
 
 	/**
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
 	protected function ensureExistingVersion(
-		string $version,
+		VersionId $versionId,
 		string $lang
 	): void {
-		if ($this->exists($version, $lang) !== true) {
-			throw new NotFoundException('Version "' . $version . ' (' . $lang . ')" does not already exist');
+		if ($this->exists($versionId, $lang) !== true) {
+			throw new NotFoundException('Version "' . $versionId->value . ' (' . $lang . ')" does not already exist');
```

Comment:
Since we implement `VersionId` as `Stringable`, the `->value` part is not needed.